### PR TITLE
Automation of pipeline-runs | pipelines-plugin 

### PR DESCRIPTION
**Pull request by SparshKesari**
**Description:**

- Automation of pipeline-runs feature file.

**Jira Id:** [ODC-6630](https://issues.redhat.com/browse/ODC-6630)

**Pre-conditions/setup:**

**Checks for approving Epic scenarios Automation PR:**

- [x] Execute the @to-do tagged gherkin scripts manually
- [x] Convert the @to-do gherkin scripts to cypress automation scripts
- [x] Once scripts are automated, replace tag @to-do with @epic-number
- [x] Execute the scripts in Remote cluster

**Refactoring criteria for approving Automation PR:**

- [ ] update the test cases count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)

**Execution Commands:**
Example: 
```
export NO_HEADLESS=true && export CHROME_VERSION=$(/usr/bin/google-chrome-stable --version)
    BRIDGE_KUBEADMIN_PASSWORD=YH3jN-PRFT2-Q429c-5KQDr
    BRIDGE_BASE_ADDRESS=https://console-openshift-console.apps.dev-svc-4.8-042801.devcluster.openshift.com
    export BRIDGE_KUBEADMIN_PASSWORD
    export BRIDGE_BASE_ADDRESS
    oc login -u kubeadmin -p $BRIDGE_KUBEADMIN_PASSWORD
    oc apply -f ./frontend/integration-tests/data/htpasswd-secret.yaml
    oc patch oauths cluster --patch "$(cat ./frontend/integration-tests/data/patch-htpasswd.yaml)" --type=merge
    ./test-cypress.sh -p pipelines

```

**Screen shots**:

image

**Browser conformance**:

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts [hunk 1]
```diff
@@ -473,3 +479,99 @@ Then(
...
+When('user goes to failed pipeline run of pipeline {string}', (pipelineName: string) => {
+  navigateTo(devNavigationMenu.Pipelines);
+  pipelinesPage.search(pipelineName);
+  cy.get(`[data-test-id^="${pipelineName}-"]`).click();
+});
+
+When('user opens pipeline run details', () => {
+  cy.get('.pf-c-breadcrumb').should('include.text', 'PipelineRun details');
+});
+Then('user can see status as Failure', () => {
```
Comment (sanketpathak) [inline frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts:506]: Give proper spacing

Comment (sanketpathak) [inline frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts:506]: Give proper spacing

File: frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts [hunk 2]
```diff
@@ -473,3 +479,99 @@ Then(
...
+});
+
+Given('user has passed pipeline run', () => {
+  cy.exec(
+    `oc apply -f testData/pipelines-workspaces/sum-three-pipeline.yaml -n ${Cypress.env(
+      'NAMESPACE',
+    )}`,
+  );
+});
+When(
```
Comment (sanketpathak) [inline frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts:532]: Check the spacing in the complete file

Comment (SparshKesari) [reply frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts]: @sanketpathak 👍

**Comment by sanketpathak** [comment]
Test runs are green
/lgtm

File: frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts [hunk 3]
```diff
@@ -473,3 +479,108 @@ Then(
     modal.shouldBeClosed();
   },
 );
+
+Given('user is at pipeline page in developer perspective', () => {
+  perspective.switchTo(switchPerspective.Developer);
```
Comment (jrichter1) [inline frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts:484]: this should also ensure the user is at the pipelines page, not just dev perspective

File: frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts [hunk 4]
```diff
@@ -473,3 +479,108 @@ Then(
...
+When('user opens pipeline run details', () => {
+  cy.get('.pf-c-breadcrumb').should('include.text', 'PipelineRun details');
+});
+
+Then('user can see status as Failure', () => {
+  cy.get(pipelineRunsPO.pipelineRunsTable.status).should('include.text', 'Failed');
+});
+
+Then('user can view failure message under Message heading', () => {
+  cy.get('.odc-pipeline-run-details__customDetails').within(() => {
```
Comment (jrichter1) [inline frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts:512]: this selector would look better in a page object, actually lets get as many selectors that aren't parameterized into a page object as we can

File: frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts [hunk 5]
```diff
@@ -473,3 +479,108 @@ Then(
...
+  cy.get('[role="grid"]').should('contain.text', 'Name');
+  cy.get('[role="grid"]').should('contain.text', 'Value');
+});
+
+When(
+  'a node with an associated pipeline {string} that has failed is present',
+  (pipelineName: string) => {
+    topologyPage.componentNode(pipelineName).should('be.visible');
+  },
+);
```
Comment (jrichter1) [inline frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts:556]: - this is a `Given` step
- shouldn't we verify the node shows a failed pipeline here?

File: frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts [hunk 6]
```diff
@@ -473,3 +479,108 @@ Then(
...
+
+Then('user will also see the log snippet', () => {
+  cy.get('.ocs-log-snippet__log-snippet').should('be.visible');
+});
+
+Given('user is at Topology page', () => {
+  navigateTo(devNavigationMenu.Topology);
+  app.waitForLoad();
+  topologyPage.verifyTopologyPage();
+});
```
Comment (jrichter1) [inline frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipelines-runs.ts:586]: This step should already be implemented, possibly in a different package

**Comment by jrichter1** [comment]
This PR only affects our e2e tests.
/label px-approved
/label docs-approved
/label qe-approved

**Pull request merged**